### PR TITLE
Add support for updated events

### PR DESCRIPTION
**Pull request by shahthepro**
To be deployed immediately after https://github.com/OriginProtocol/origin-dollar/pull/459

### CODE REVIEW

File: eagleproject/core/blockchain.py [hunk 1]
```diff
@@ -543,13 +543,16 @@ def maybe_store_stake_withdrawn_record(log, block):
...
+
     staked = OgnStaked(
         tx_hash=tx_hash,
         log_index=log_index,
         block_time=block.block_time,
         user_address="0x" + log["topics"][1][-40:],
-        is_staked=log["topics"][0] == SIG_EVENT_STAKED,
+        is_staked=is_staked_event,
         amount=int(_slot(log["data"], 0), 16) / 1e18,
+        staked_amount=int(_slot(log["data"], 1), 16) / 1e18 if not is_staked_event else 0,
```
Comment (franckc) [inline eagleproject/core/blockchain.py:555]: should we also store the staked_duration? I think that will be helpful for the business to run some analysis on user's staking behavior.

File: eagleproject/core/blockchain.py [hunk 2]
```diff
@@ -54,8 +54,8 @@
 
 TRANSFER = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
 
-SIG_EVENT_STAKED = encode_hex(keccak(b"Staked(address,uint256)"))
-SIG_EVENT_WITHDRAWN = encode_hex(keccak(b"Withdrawn(address,uint256)"))
+SIG_EVENT_STAKED = encode_hex(keccak(b"Staked(address,uint256,uint256,uint256)"))
+SIG_EVENT_WITHDRAWN = encode_hex(keccak(b"Withdrawn(address,uint256,uint256)"))
```
Comment (mikeshultz) [inline eagleproject/core/blockchain.py:58]: How about we add support for both the legacy and new events?  There's already the old events on the chain that would be ignored if we change the sig and that's that.

**Review by mikeshultz — COMMENTED** [review]
Think we should probably be backward compatible on these, lest we can't recreate the DB in the future.

File: eagleproject/core/blockchain.py (hunk 1 above)
Comment (shahthepro) [reply eagleproject/core/blockchain.py]: Will add it

File: eagleproject/core/blockchain.py (hunk 2 above)
Comment (shahthepro) [reply eagleproject/core/blockchain.py:61]: I have made that change but the code is untested though. Let me check if I can run these triggers locally

File: eagleproject/notify/triggers/staking_stake.py [hunk 3]
```diff
@@ -24,17 +27,50 @@ def run_trigger(new_logs):
...
+        elif ev.topic_0 == SIG_EVENT_STAKED
+            (amount,duration,rate,) = decode_single('(uint256,uint256,uint256)', decode_hex(ev.data))
+
+            duration_dt = timedelta(seconds=duration)
+
+            apy_multiple = Decimal(DAYS_365_SECONDS) / Decimal(duration)
+            apy = round((Decimal(rate) / Decimal(1e18)) * apy_multiple * Decimal(100), 1)
+
+            events.append(
+                event.normal(
```
Comment (mikeshultz) [inline eagleproject/notify/triggers/staking_stake.py:54]: `event_normal`

File: eagleproject/core/blockchain.py [hunk 4]
```diff
@@ -543,13 +546,20 @@ def maybe_store_stake_withdrawn_record(log, block):
...
+
     staked = OgnStaked(
         tx_hash=tx_hash,
         log_index=log_index,
         block_time=block.block_time,
         user_address="0x" + log["topics"][1][-40:],
-        is_staked=log["topics"][0] == SIG_EVENT_STAKED,
+        is_staked=is_staked_event,
         amount=int(_slot(log["data"], 0), 16) / 1e18,
+        staked_amount=int(_slot(log["data"], 1), 16) / 1e18 if is_withdrawn_event else 0,
```
Comment (mikeshultz) [inline eagleproject/core/blockchain.py:560]: Isn't this the same as using `amount` and `is_staked`?  What's gained by storing this?

File: eagleproject/core/blockchain.py (hunk 2 above)
Comment (mikeshultz) [reply eagleproject/core/blockchain.py:61]: If you want a hand testing, or getting setup, hit me up in discord.

File: eagleproject/core/blockchain.py (hunk 4 above)
Comment (mikeshultz) [reply eagleproject/core/blockchain.py:560]: ah, I guess you're going for having the Stake/Withdraw amounts line up and not be negative when all is said and done.  I might suggest a different name for clarity but I see what you're going for now.  n/m

**Comment by shahthepro** [comment]
@mikeshultz @franckc I tested this with the updated contract events, works fine for me. Merging this now, can you help deploy this?

**Pull request merged**